File: php/05-functions/scope.php
<?php
declare(strict_types=1);

$message = "I live in the global scope";

function showScope(): void
{
    // This local $message is separate from the global one
    $message = "I live inside the function";
    echo $message . "\n";

    // To access the global variable, declare it
    global $message;
    echo $message . "\n";
}

// A static variable keeps its value across calls
function counter(): int
{
    static $count = 0;
    $count++;
    return $count;
}

showScope();
echo "Call 1: " . counter() . "\n";
echo "Call 2: " . counter() . "\n";
echo "Call 3: " . counter() . "\n";

File: python/05-functions/scope.py
# Local vs global scope

counter = 0  # module-level (global) variable

def increment():
    global counter      # rebind the global, not a new local
    counter += 1
    return counter

def local_demo():
    counter = 100       # a new local variable that shadows the global
    return counter

print(increment())   # 1
print(increment())   # 2
print(local_demo())  # 100
print(counter)       # 2 - local_demo never touched the global
